CRM/Core/Form/EntityTrackingTrait.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * This trait provides a form of lazy loading for forms.
+ *
+ * It is intended to reduce the need for forms to pass values around while avoiding
+ * constantly reloading them from the database.
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC https://civicrm.org/licensing
+ */
+trait CRM_Core_Form_EntityTrackingTrait {
+
+  /**
+   * Array of loaded entities.
+   *
+   * These are stored with keys that are consistent with apiv4 style parameters.
+   *
+   * @var array
+   */
+  private $entities = [];
+
+  public function setEntity($entity, $identifier, $values): void {
+    $this->entities[$entity][$identifier] = $values;
+  }
+
+  /**
+   * Get the value for a property of an entity, loading from the database if needed.
+   *
+   * Permissions are not applied to the api call.
+   *
+   * @param string $entity
+   * @param int $id
+   * @param string $key
+   *
+   * @api supported for use outside of core. Will not change in a point release.
+   *
+   * @return mixed
+   */
+  public function getEntityValue(string $entity, int $id, string $key) {
+    if (!isset($this->entities[$entity][$id]) || !array_key_exists($key, $this->entities[$entity][$id])) {
+      $this->loadEntity($entity, $id);
+    }
+    return $this->entities[$entity][$id][$key];
+  }
+
+  /**
+   * Get a value from a participant record.
+   *
+   * This function requires that the form implements `getParticipantID()`.
+   *
+   * @param string $fieldName
+   * @param int|null $id If not provided getParticipantID() is called.
+   *
+   * @api supported for use outside of core. Will not change in a point release.
+   *
+   * @return mixed
+   */
+  public function getParticipantValue(string $fieldName, ?int $id = NULL) {
+    $id = $id ?? $this->getParticipantID();
+    return $this->getEntityValue('Participant', $id, $fieldName);
```

Comment:
@eileenmcnaughton this doesn't seem to be exactly the same as the original as the original could find the value in `participant_$fieldname` if it wasn't present in $fieldName` right?

Reply:
@seamuslee001 true - but the actual usage of this function means that isn't an issue - 

![image](https://github.com/civicrm/civicrm-core/assets/336308/927f0be8-95de-4a27-98c4-050194f9379a)